compiler/src/dmd/safe.d

```diff
@@ -317,6 +317,16 @@ bool checkUnsafeDotExp(Scope* sc, Expression e, Identifier id, int flag)
     return false;
 }
 
+/**************************************
+ * Safer D adds safety checks to functions with the default
+ * trust setting.
+ */
+bool isSaferD(FuncDeclaration fd)
+{
+    return fd.type.toTypeFunction().trust == TRUST.default_ &&
+           global.params.safer == FeatureState.enabled;
```

Comment:
I would prefer this checking a scope flag rather than a global variable

Reply:
This will likely be limited to newer editions, so it'll have to check it at the module level. Which eliminates the global flag.